dispatch/tests/test_api_pages.py

```diff
@@ -0,0 +1,226 @@
+          "title": "Test Page",
+          "slug": "test-page",
+          "snippet": "This is a test snippet",
+          "content": [
+            {
+              "type": "paragraph",
+              "data": "This is some paragraph text"
+            }
+          ]
+        }
+
+        return self.client.post(url, data, format='json')
+
+    def test_create_page_unauthorized(self):
+        """
+        Create page while unauthorized should fail with unauthenticated request
+        """
+        # Clear authentication credentials
+        self.client.credentials()
+
+        url = reverse('api-pages-list')
+
+        data = {
+          "title": "Test Page",
+          "slug": "test-page",
+          "snippet": "This is a test snippet",
+          "content": [
+            {
+              "type": "paragraph",
+              "data": "This is some paragraph text"
+            }
+          ]
+        }
+
+        response = self.client.post(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_create_empty_page(self):
+        """
+        Create page should fail with empty payload
+        """
+
+        url = reverse('api-pages-list')
+
+        self._create_page()
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_create_incomplete_page(self):
+        """
+        Creat Page should fail with missing required fields
+        """
+
+        url = reverse('api-pages-list')
+
+        # Page missing content, snippet, slug
+        data = {
+            "title": "Test Page",
+        }
+
+        response = self.client.post(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        # Check Data
+        self.assertTrue('content_json' in response.data)
+        self.assertTrue('snippet' in response.data)
+        self.assertTrue('slug' in response.data)
+
+    def test_create_page(self):
+        """
+        Ensure we can make a page
+        """
+
+        response = self._create_page()
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # Check Data
+        self.assertEqual(response.data['title'], 'Test Page')
+        self.assertEqual(response.data['slug'], 'test_page')
+        self.assertEqual(response.data['snippet'], "This is a test snippet")
+        self.assertEqual(response.data['content'][0]['type'], 'paragraph')
+        self.assertEqual(response.data['content'][0]['data'], 'This is some paragraph text')
+
+    def test_update_fields_page(self):
+        """
+        Should be able to update the title, slug, snippet of the page
+        """
+
+        # First make a page
+        page = self._create_page()
+
+        # Generate detail url
+        url = reverse('api-pages-detail', args=[page.data['id']])
+
+        # Change the fields
+        new_data = {
+          "title": "New Test Page",
+          "slug": "New test-page",
```

Comment:
A slug can't contain spaces (not a valid URL). This should be changed to `'new-test-page'`